File: Frontend/src/types/owner-application.ts
export type OwnerApplicationStatus =
  | "PENDING_APPROVAL"
  | "APPROVED"
  | "REJECTED";

export interface OwnerApplyRequest {
  businessName: string;
  contactPhone: string;
  address: string;
  city: string;
  district: string;
  description?: string;
  businessLicenseNumber?: string;
}

export interface OwnerApplicationMe {
  id: string;
  businessName: string;
  contactPhone: string;
  address: string;
  city: string;
  district: string;
  description?: string | null;
  businessLicenseNumber?: string | null;
  status: OwnerApplicationStatus;
  rejectionReason?: string | null;
  approvedAt?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface AdminOwnerApplicationListItem {
  id: string;
  userId: string;
  userFullName: string;
  userEmail: string;
  userPhone?: string | null;
  businessName: string;
  contactPhone: string;
  city: string;
  district: string;
  status: OwnerApplicationStatus;
  createdAt: string;
}

export interface AdminOwnerApplicationListResponse {
  items: AdminOwnerApplicationListItem[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface AdminOwnerApplicationDetail {
  id: string;
  userId: string;
  userFullName: string;
  userEmail: string;
  userPhone?: string | null;
  userRole: string;
  businessName: string;
  contactPhone: string;
  address: string;
  city: string;
  district: string;
  description?: string | null;
  businessLicenseNumber?: string | null;
  status: OwnerApplicationStatus;
  rejectionReason?: string | null;
  approvedAt?: string | null;
  createdAt: string;
  updatedAt: string;
}

File: Backend/MATCHOP.API/Entities/OwnerApplication.cs
using MATCHOP.API.Enums;

namespace MATCHOP.API.Entities;

public class OwnerApplication
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }

    public string BusinessName { get; set; } = null!;
    public string ContactPhone { get; set; } = null!;
    public string Address { get; set; } = null!;
    public string City { get; set; } = null!;
    public string District { get; set; } = null!;
    public string? Description { get; set; }
    public string? BusinessLicenseNumber { get; set; }

    public OwnerApplicationStatus Status { get; set; } = OwnerApplicationStatus.PENDING_APPROVAL;
    public string? RejectionReason { get; set; }
    public DateTime? ApprovedAt { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
}
